src/Compilers/CSharp/Portable/Symbols/LocalDeclarationKind.cs

```diff
@@ -49,5 +49,16 @@ internal enum LocalDeclarationKind : byte
         /// The variable that captures the result of a pattern matching operation like "i" in "expr is int i"
         /// </summary>
         PatternVariable,
+
+        /// <summary>
+        /// User variable declared by a declaration expression in the left-hand-side of a deconstruction assignment.
+        /// </summary>
+        Deconstruction,
+
+        /// <summary>
+        /// User variable declared by a declaration expression that is not a deconstruction. This occurs for an
+        /// out variable declaration, and as a result of error recovery in incorrect code.
+        /// </summary>
+        DeclarationExpression,
```

Comment:
Merging the kinds as you suggest makes it impossible, for example, to determine by inspection whether or not a variable is an *expression variable* (a subset of the kinds that can be declared in an expression). Such concepts are useful and I prefer to keeps things factored in a way that reflects both the syntax and the structure of the compiler.